<a href="https://colab.research.google.com/github/malcommathela/Data-Analysis/blob/main/campus_placement_assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Campus Placement Assignment
**Prepared by: Malcom Mathela — for the Campus Placement Cell**

This notebook follows the 3 guided stages:
1. **Load:** Ingest `campus_data.csv` and inspect the shape
2. **Clean & Aggregate :** Fill missing CGPA, department-wise averages, sorting
3. **Filter & Transform :** Filter top scorers and flag interview eligibility

## Setup: Run this cell first to create the sample dataset

In [1]:
import pandas as pd
import numpy as np

data = {
    "StudentID": [101, 102, 103, 104, 105, 106],
    "Name": ["Aarav", "Bhavya", "Charan", "Divya", "Eshwar", "Farhan"],
    "Department": ["CSE", "ECE", "CSE", "IT", "ECE", "IT"],
    "CGPA": [8.2, 7.5, np.nan, 9.1, 6.8, np.nan],
    "CRT Score": [88, 62, 79, 95, 55, 84]
}
pd.DataFrame(data).to_csv("campus_data.csv", index=False)

## Stage 1 — Load
Ingest `campus_data.csv` using `pd.read_csv()` and print the dataset shape.

In [2]:
df = pd.read_csv("campus_data.csv")
print("Dataset shape (rows, columns):", df.shape)
df

Dataset shape (rows, columns): (6, 5)


,StudentID,Name,Department,CGPA,CRT Score
0,101,Aarav,CSE,8.2,88
1,102,Bhavya,ECE,7.5,62
2,103,Charan,CSE,NaN,79
3,104,Divya,IT,9.1,95
4,105,Eshwar,ECE,6.8,55
5,106,Farhan,IT,NaN,84


## Stage 2 — Clean & Aggregate
1. Fill missing CGPA values with the column average using `.fillna()`
2. Calculate the average CodingScore department-wise using `.groupby()`
3. Sort students by CodingScore from highest to lowest

In [3]:
# 2.1 Fill missing CGPA with the column average
df["CGPA"] = df["CGPA"].fillna(df["CGPA"].mean())
print("Missing CGPA values remaining:", df["CGPA"].isna().sum())
df

Missing CGPA values remaining: 0


,StudentID,Name,Department,CGPA,CRT Score
0,101,Aarav,CSE,8.2,88
1,102,Bhavya,ECE,7.5,62
2,103,Charan,CSE,7.9,79
3,104,Divya,IT,9.1,95
4,105,Eshwar,ECE,6.8,55
5,106,Farhan,IT,7.9,84


In [4]:
# 2.2 Average CodingScore department-wise
dept_avg = df.groupby("Department")["CRT Score"].mean()
print("Department-wise average CodingScore:")
dept_avg

Department-wise average CodingScore:


,CRT Score
Department,
CSE,83.5
ECE,58.5
IT,89.5


In [5]:
# 2.3 Sort by CodingScore, highest to lowest
df_sorted = df.sort_values(by="CRT Score", ascending=False).reset_index(drop=True)
df_sorted

,StudentID,Name,Department,CGPA,CRT Score
0,104,Divya,IT,9.1,95
1,101,Aarav,CSE,8.2,88
2,106,Farhan,IT,7.9,84
3,103,Charan,CSE,7.9,79
4,102,Bhavya,ECE,7.5,62
5,105,Eshwar,ECE,6.8,55


## Stage 3 — Filter & Transform
1. Filter students with CodingScore >= 80
2. Add a new column `Interview_Eligible` set to `"Yes"` for all qualifying students

In [6]:
# 3.1 Filter students with CodingScore >= 80
eligible = df[df["CRT Score"] >= 80].copy()

# 3.2 Add the Interview_Eligible column
eligible["Interview_Eligible"] = "Yes"

print("Number of interview-eligible students:", len(eligible))
eligible

Number of interview-eligible students: 3


,StudentID,Name,Department,CGPA,CRT Score,Interview_Eligible
0,101,Aarav,CSE,8.2,88,Yes
3,104,Divya,IT,9.1,95,Yes
5,106,Farhan,IT,7.9,84,Yes


## Summary for the Placement Cell

| Metric | Result |
|---|---|
| Dataset shape | (6, 5) |
| Missing CGPA values filled | 2 (filled with column average = 7.90) |
| Department avg CodingScore | CSE: 83.5, ECE: 58.5, IT: 89.5 |
| Top scorer | Divya (IT) — 95 |
| Interview-eligible (>= 80) | 3 students — Divya, Aarav, Farhan |

**Recommendation:** Prioritize IT and CSE candidates for the next interview round; ECE may need additional CRT preparation (avg 58.5).